File: modules/guides/examples/index/build-index.js
// tag::defer[]
const indexMgr = cluster.queryIndexes();
await indexMgr.createPrimaryIndex('bucket_name', {deferred: true})
await indexMgr.createIndex('bucket_name', 'ix_name', ['name'], {deferred: true})
await indexMgr.createIndex('bucket_name', 'ix_email', ['email'], {deferred: true})
// end::defer[]

// tag::build[]
await indexMgr.buildDeferredIndexes('bucket_name')
await indexMgr.watchIndexes('bucket_name', ['ix_name', 'ix_email', '#primary'], 2000)
// end::build[]

File: modules/guides/examples/index/build-index.java
// tag::defer[]
QueryIndexManager indexManager = cluster.queryIndexes();

indexManager.createPrimaryIndex(bucketName,
    CreatePrimaryQueryIndexOptions.createPrimaryQueryIndexOptions().deferred(true));
indexManager.createIndex(bucketName, "ix_name", Collections.singletonList("name"),
    CreateQueryIndexOptions.createQueryIndexOptions().deferred(true));
indexManager.createIndex(bucketName, "ix_email", Collections.singletonList("preferred_email"),
    CreateQueryIndexOptions.createQueryIndexOptions().deferred(true));
// end::defer[]

// tag::build[]
indexManager.buildDeferredIndexes(bucketName);
indexManager.watchIndexes(bucketName, Arrays.asList("ix_name", "ix_email"), Duration.ofMinutes(5));
// end::build[]